app/controllers/v0/preferences_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module V0
+  class PreferencesController < ApplicationController
+    before_action :set_preference
+
+    def show
+      render json: @preference
+    end
+
+    private
+
+    def set_preference
+      @preference = Preference.find_by(code: preference_params['code'])
```

Comment:
Instead of pulling the `PreferenceChoices` in the serializer, we'll need to create a method on `app/models/preference.rb` that yields a`Preference` and its associated `PreferenceChoices`.

Reason being, we need to cache all of these records _prior_ to arriving at the serializer.  That way, when the controller calls for the `Preference` and the `PreferenceChoices`, we can first check the cache, and if they're not in there, we can fetch & cache them.  

You know what I mean?

Reply:
Addressed in https://github.com/department-of-veterans-affairs/vets-api/pull/2262/commits/463ca689852dd4ca269a3161a6e40ca0763e9d3c